oio/account/backend.py

```diff
@@ -418,16 +418,20 @@ def _raw_listing(self, account_id, limit, marker, end_marker, delimiter,
                     if dir_name != orig_marker:
                         results.append([dir_name, 0, 0, 1, 0])
                     break
-                results.append([container_id, 0, 0, 0, 0])
+                if not buckets or (buckets
+                                   and "+segments" not in container_id
+                                   and "%2F" not in container_id):
```

Comment:
We could simplify the condition:
```
not buckets or ('+segments' not in container_id and '%2F' not in container_id)
```

Reply:
Technically, bucket name should respect https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html#bucketnamingrules

Reply:
We could use a regex: https://stackoverflow.com/questions/50480924/regex-for-s3-bucket-name

Reply:
Are you sure to use this regex `(?=^.{3,63}$)(?!^(\d+\.)+\d+$)(^(([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])\.)*([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])$)` ?

Reply:
Yes, if you explain each part.
This regex is used to check all cases. But your conditions do not check the following 2 cases:
- Bucket names must be a series of one or more labels. Adjacent labels are separated by a single period (.). Bucket names can contain lowercase letters, numbers, and hyphens. Each label must start and end with a lowercase letter or a number.
- Bucket names must not be formatted as an IP address (for example, 192.168.5.4).